lib/src/casing.rs

```diff
@@ -82,6 +86,10 @@ pub(crate) fn mk_name(s: &str) -> Option<Name> {
             }
         }
     }
-    let camel = mk_ident(std::str::from_utf8(&buf).ok()?);
+    // SAFETY: `validate` ensures that all source bytes match `a-z0-9_-`,
+    // and all subsequent writes use bytes that also match said pattern
+    let camel = unsafe {
```

Comment:
You might appreciate https://www.ecorax.net/tightness/
and https://crates.io/crates/tightness

For creating a type that only holds those characters xD

Reply:
Looks cool, but it seems to insert runtime checks on mutation (which I tried to avoid here). Definitely a neat crate, though.